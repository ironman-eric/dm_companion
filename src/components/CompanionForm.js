import { useState } from "react"
import ClassContainer from './ClassContainer';
import SpeciesContainer from './SpeciesContainer';
import CompanionResult from './CompanionResult';

function CompanionForm() {
  const [submitted, setSubmitted] = useState(0);

  const [form, setForm] = useState({
    dndClass: '',
    species: ''
  });

  const handleChange = (event) => {
    setForm(Object.assign(form, { [event.target.name]: event.target.value }));
  };

  const validate = () => {
    if (form.dndClass.length === 0 || form.species.length === 0) {
      alert('Invalid Form, Please make some selections')
      return false
    }
    return true
  }

  const handleSubmit = (event) => {    
    event.preventDefault();
    if (validate()) {
      setSubmitted(submitted + 1);
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <div class="container mx-auto">
        <div class="grid grid-cols-2 gap-1">
          <ClassContainer onChangeValue={handleChange} />
          <SpeciesContainer onChangeValue={handleChange} />
        </div>
        <div class="flex items-center justify-between">
          <button type="submit" class="mb-2 block w-full rounded bg-primary px-6 pb-2 pt-2.5 text-xs font-medium uppercase leading-normal text-white shadow-[0_4px_9px_-4px_#3b71ca] transition duration-150 ease-in-out hover:bg-primary-600 hover:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)] focus:bg-primary-600 focus:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)] focus:outline-none focus:ring-0 active:bg-primary-700 active:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)] dark:shadow-[0_4px_9px_-4px_rgba(59,113,202,0.5)] dark:hover:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.2),0_4px_18px_0_rgba(59,113,202,0.1)] dark:focus:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.2),0_4px_18px_0_rgba(59,113,202,0.1)] dark:active:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.2),0_4px_18px_0_rgba(59,113,202,0.1)]">
            Action
          </button>
        </div>
        <div>
          {submitted > 0 &&
            <CompanionResult data={form} />
          }
        </div>
      </div>
    </form>
  );
}

export default CompanionForm;
