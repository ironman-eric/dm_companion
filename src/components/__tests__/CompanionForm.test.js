import React from "react";
import renderer from "react-test-renderer";
import CompanionForm from "../CompanionForm";

describe("CompanionForm", () => {
  it("Matches DOM Snapshot", () => {
    const domTree = renderer.create(<CompanionForm />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
