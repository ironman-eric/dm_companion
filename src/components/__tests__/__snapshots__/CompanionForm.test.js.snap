// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanionForm Matches DOM Snapshot 1`] = `
<div
  className="p-6 bg-sky-100 rounded"
>
  <div
    className="flex items-center justify-center font-black m-3 mb-12"
  >
    <form
      className="flex flex-col justify-center"
      onSubmit={[Function]}
    >
      <div
        className="mb-4"
      >
        <label
          className="block text-gray-700 text-sm font-bold mb-2"
        >
          Select a Class
        </label>
        <span
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
        >
          <input
            name="dndClass"
            onChange={[Function]}
            type="radio"
            value="Barbarian"
          />
            
          Barbarian
        </span>
        <span
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
        >
          <input
            name="dndClass"
            onChange={[Function]}
            type="radio"
            value="Bard"
          />
            
          Bard
        </span>
        <span
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
        >
          <input
            name="dndClass"
            onChange={[Function]}
            type="radio"
            value="Cleric"
          />
            
          Cleric
        </span>
        <span
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
        >
          <input
            name="dndClass"
            onChange={[Function]}
            type="radio"
            value="Fighter"
          />
            
          Fighter
        </span>
        <span
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
        >
          <input
            name="dndClass"
            onChange={[Function]}
            type="radio"
            value="Monk"
          />
            
          Monk
        </span>
        <span
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
        >
          <input
            name="dndClass"
            onChange={[Function]}
            type="radio"
            value="Ranger"
          />
            
          Ranger
        </span>
        <span
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
        >
          <input
            name="dndClass"
            onChange={[Function]}
            type="radio"
            value="Rogue"
          />
            
          Rogue
        </span>
        <span
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
        >
          <input
            name="dndClass"
            onChange={[Function]}
            type="radio"
            value="Wizard"
          />
            
          Wizard
        </span>
      </div>
      <div
        className="flex items-center justify-between"
      >
        <button
          className="px-8 py-3 block w-full text-white bg-blue-600 rounded focus:outline-none disabled:opacity-25"
          disabled={true}
          type="submit"
        >
          Action
        </button>
      </div>
      <div
        className="flex items-center justify-between"
      />
    </form>
  </div>
</div>
`;
